pkg/controller/atlasinstance/atlasinstance_controller.go

```diff
@@ -0,0 +1,387 @@
+			Labels: map[string]string{
+				"created-by": "atlas-operator",
+			},
+		},
+		Spec: v1.AtlasProjectSpec{
+			Name:                data.ProjectName,
+			ConnectionSecret:    &v1.ResourceRef{Name: secretName},
+			ProjectIPAccessList: []project.IPAccessList{},
+		},
+	}
+}
+
+// getAtlasClusterSpec returns the spec for the desired cluster
+func getAtlasClusterSpec(instance *dbaas.MongoDBAtlasInstance, data *InstanceData) *v1.AtlasClusterSpec {
+	var providerSettingsSpec *v1.ProviderSettingsSpec
+	if data.InstanceSizeName == "M0" || data.InstanceSizeName == "M2" || data.InstanceSizeName == "M5" {
+		// See Atlas documentation https://docs.atlas.mongodb.com/reference/api/clusters-create-one/
+		providerSettingsSpec = &v1.ProviderSettingsSpec{
+			InstanceSizeName:    data.InstanceSizeName,
+			BackingProviderName: data.ProviderName,
+			ProviderName:        provider.ProviderName("TENANT"),
+			RegionName:          data.RegionName,
+		}
+	} else {
+		providerSettingsSpec = &v1.ProviderSettingsSpec{
+			InstanceSizeName: data.InstanceSizeName,
+			ProviderName:     provider.ProviderName(data.ProviderName),
+			RegionName:       data.RegionName,
+		}
+	}
+	return &v1.AtlasClusterSpec{
+		Project:          v1.ResourceRefNamespaced{Name: data.ProjectName, Namespace: instance.Namespace},
+		Name:             data.ClusterName,
+		ProviderSettings: providerSettingsSpec,
+	}
+}
+
+// getOwnedAtlasCluster returns an AtlasCluster object owned by the instance
+func getOwnedAtlasCluster(instance *dbaas.MongoDBAtlasInstance, data *InstanceData) *v1.AtlasCluster {
+	return &v1.AtlasCluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      data.ClusterName,
+			Namespace: instance.Namespace,
+			Labels: map[string]string{
+				"managed-by":      "atlas-operator",
+				"owner":           instance.Name,
+				"owner.kind":      instance.Kind,
+				"owner.namespace": instance.Namespace,
+			},
+			OwnerReferences: []metav1.OwnerReference{
+				{
+					UID:                instance.GetUID(),
+					APIVersion:         dbaas.GroupVersion.Identifier(),
+					BlockOwnerDeletion: ptr.BoolPtr(false),
+					Controller:         ptr.BoolPtr(true),
+					Kind:               "MongoDBAtlasInstance",
+					Name:               instance.Name,
+				},
+			},
+		},
+	}
+}
+
+func getInstanceData(log *zap.SugaredLogger, inst *dbaas.MongoDBAtlasInstance) (*InstanceData, error) {
+	if len(inst.Spec.Name) == 0 {
+		log.Errorf("Missing %v", dbaas.ClusterNameKey)
+		return nil, fmt.Errorf("missing %v", dbaas.ClusterNameKey)
+	}
+	projectName, ok := inst.Spec.OtherInstanceParams[dbaas.ProjectNameKey]
+	if !ok || len(projectName) == 0 {
+		log.Errorf("Missing %v", dbaas.ProjectNameKey)
+		return nil, fmt.Errorf("missing %v", dbaas.ProjectNameKey)
+	}
+	provider := strings.ToUpper(inst.Spec.CloudProvider)
+	if len(provider) == 0 {
+		provider = "AWS"
+		log.Infof("%v is missing, default value of AWS is used", dbaas.CloudProviderKey)
+	}
+	region := inst.Spec.CloudRegion
```

Comment:
We are supposed to pass these values hard coded for this release, as per the trial _version. I mean should be the default region/provider/etc?

Reply:
For the trial release, the user does not need to pass in region, cloud provider and instance size. When these values are missing, the operator sets the default values for them.